source/dub/dub.d

```diff
@@ -277,28 +277,104 @@ class Dub {
 
 		The script above can be invoked with "dub --single test.d".
 	*/
+		struct Parser {
+			static typeof(this) opCall(string sc) {
+				Parser ret;
+				ret.sourcecode = sc;
+				ret.popFront;
+				return ret;
+			}
+			string sourcecode;
+			string front;
+			bool empty;
+			char inComment = '\0';
+			bool hadFileName;
+			bool hadData;
+			int depth;
+			void popFront() {
+				import std.uni : isWhite;
+				if (empty) return;
+				for (;sourcecode.length > 0;) {
+					if (sourcecode[0].isWhite) {
+						sourcecode = sourcecode[1..$];
```

Comment:
It maybe doesn't matter in practice, but since this compares a single byte, `std.ascii : isWhite` would be the right choice here. A value greater than `0x7F` could be erroneously matched as a whitespace character (a non-breaking space for example).